Check for overflow when validating FEConvolveMatrix kernelSize

jon-b-apple opened the pull request:
#### 198b72f17045e58ccb4e2015286b499053757f0f
<pre>
Check for overflow when validating FEConvolveMatrix kernelSize
<a href="https://bugs.webkit.org/show_bug.cgi?id=297526">https://bugs.webkit.org/show_bug.cgi?id=297526</a>
<a href="https://rdar.apple.com/158573872">rdar://158573872</a>

Reviewed by NOBODY (OOPS!).

This patch adds a canOverflow method on IntSize, which lets us test for overflow without crashing for the purpose of validation.
It adopts this method in the IPC validation, which avoids a crash in the area getter when using Checked with CrashOnOverflow.

* Source/WebCore/platform/graphics/IntSize.h:
(WebCore::IntSize::canOverflow const):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/198b72f17045e58ccb4e2015286b499053757f0f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129414 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/1671 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40253 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/136791 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/80826 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/a024aa1b-0f45-43e4-aa73-768d8f065776) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/131285 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/1602 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/1547 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/98562 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/66456 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/5cdcf1c4-5f1e-42ad-8b8b-4d0a27d1bc98) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132361 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/1248 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/115910 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79213 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/54d06d44-9e50-4365-9916-cc944349ab24) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/1171 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/34041 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/80068 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/109625 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/34541 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139265 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/1461 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1405 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/107088 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/1503 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112254 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/106931 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/1190 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/30774 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/54129 "The change is no longer eligible for processing.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/1532 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/64895 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1349 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/1386 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1454 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
EWS run on current version of this PR (hash https://github.com/WebKit/WebKit/commit/2aec8cf5fa1665fb39845e35b7ed7d6761b561f0)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129411 "Failed to checkout and rebase branch from PR 49523") | [❌ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/1668 "Failed to checkout and rebase branch from PR 49523") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40250 "Failed to checkout and rebase branch from PR 49523") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/136788 "Build is in progress. Recent messages:") | [⏳ 🛠 win ](https://ews-build.webkit.org/#/builders/Win-Build-EWS "Waiting in queue, processing has not started yet") | ⏳ 🛠 ios-apple 
| [❌ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/131282 "Failed to checkout and rebase branch from PR 49523") | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/1599 "Failed to checkout and rebase branch from PR 49523") | [❌ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/1544 "Failed to checkout and rebase branch from PR 49523") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/136788 "Build is in progress. Recent messages:") | [⏳ 🧪 win-tests ](https://ews-build.webkit.org/#/builders/Win-Build-EWS "Waiting in queue, processing has not started yet") | ⏳ 🛠 mac-apple 
| [❌ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132358 "Failed to checkout and rebase branch from PR 49523") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/155/builds/1599 "Failed to checkout and rebase branch from PR 49523") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/138/builds/40250 "Failed to checkout and rebase branch from PR 49523") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/136788 "Build is in progress. Recent messages:") | | ⏳ 🛠 vision-apple 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/155/builds/1599 "Failed to checkout and rebase branch from PR 49523") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/138/builds/40250 "Failed to checkout and rebase branch from PR 49523") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/80065 "Build is in progress. Recent messages:Printed configuration") | | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/155/builds/1599 "Failed to checkout and rebase branch from PR 49523") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/138/builds/40250 "Failed to checkout and rebase branch from PR 49523") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139262 "Build is in progress. Recent messages:Printed configuration") | | 
| | [❌ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/1458 "Failed to checkout and rebase branch from PR 49523") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/156/builds/1544 "Failed to checkout and rebase branch from PR 49523") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/139262 "Build is in progress. Recent messages:Printed configuration") | | 
| | [❌ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/1500 "Failed to checkout and rebase branch from PR 49523") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/138/builds/40250 "Failed to checkout and rebase branch from PR 49523") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/139262 "Build is in progress. Recent messages:Printed configuration") | | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-26-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/138/builds/40250 "Failed to checkout and rebase branch from PR 49523") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/54124 "Failed to checkout and rebase branch from PR 49523") | | 
| | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/1529 "Failed to checkout and rebase branch from PR 49523") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/64892 "Failed to checkout and rebase branch from PR 49523") | | | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1346 "Failed to checkout and rebase branch from PR 49523") | | | | 
| | [⏳ 🛠 watch ](https://ews-build.webkit.org/#/builders/watchOS-26-Build-EWS "Waiting in queue, processing has not started yet") | | | | 
| | [⏳ 🛠 watch-sim ](https://ews-build.webkit.org/#/builders/watchOS-26-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | | | </details>

jon-b-apple committed:
Check for overflow when validating FEConvolveMatrix kernelSize
https://bugs.webkit.org/show_bug.cgi?id=297526
rdar://158573872

Reviewed by NOBODY (OOPS!).

This patch adds a canOverflow method on IntSize, which lets us test for overflow without crashing for the purpose of validation.
It adopts this method in the IPC validation, which avoids a crash in the area getter when using Checked with CrashOnOverflow.

* Source/WebCore/platform/graphics/IntSize.h:
(WebCore::IntSize::canOverflow const):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
